streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -226,7 +231,21 @@ public GlobalStreamThread(final ProcessorTopology topology,
          * @throws StreamsException      if the store's change log does not contain the partition
          */
         void initialize() {
-            final Map<TopicPartition, Long> partitionOffsets = stateMaintainer.initialize();
+            final Map<TopicPartition, Long> partitionOffsets = new HashMap<>(stateMaintainer.initialize());
```

Comment:
Can we add test to `GlobalStreamThreadTest` covering `initialize` with this condition? I ran `./gradlew streams reportCoverage` the overall test suite hits this section, but it would be nice to have a specific unit test for this case.

Reply:
Good point! I am going to add a unit test for this case.